plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -259,6 +259,23 @@ protected function prepare_objects_query( $request ) {
 			}
 		}
 
+		if ( ! empty( $request['global_unique_id'] ) ) {
+			$global_unique_ids = explode( ',', $request['global_unique_id'] );
+			// Include the current string as a Global Unique ID too.
+			if ( 1 < count( $global_unique_ids ) ) {
+				$global_unique_ids[] = $request['global_unique_id'];
+			}
```

Comment:
The global unique ID in WooCommerce cannot contain a `,` in it, so there is no need to do this. It's adding a value that will never match.

For example, if I search for `123,456`, the `$global_unique_ids` becomes:

```php
[
    '123',
    '456',
    '123,456', // this one never will match
]
```